### ISSUE TITLE
Task: Node project & developer tooling

### ISSUE THREAD

**Issue by christiannaths** [thjones88/featsofchristian#4]
Great job so far, but your code formatting needs some attention (inconsistent spacing between lines, tabs vs spaces, 2 spaces vs 4 spaces). Everybody has different preferences for this—and even these vary by language. Good news is that there are common tools we can leverage to erase any discrepancies in this regard.

## Goals for this task

- [ ] Every time you hit save in a javascript, css, or html file, your code editor should automatically format your code

## Step 1: Work in a new branch

Recently, some changes have been approved and merged into the `master` branch, but you should always do your work in a new branch that is _based_ on the most up-to-date version of `master`.

- [ ] Switch to `master` on your local machine and do a `git pull` (this is shorthand for `git pull origin master`, where "origin" is this this remote repository on Github). This will pull down the latest changes and merge them into your local master branch. This is necessary because the changes that came from the last pull request _only_ exist on Github atm, so you need to get that code to your local machine.
- [ ] Once local `master` has been updated, do a `git checkout -b "node-tooling"`— this creates a new branch called "node_tooling" (that's what the `-b` flag denotes), and switches to that new branch (that's what the `checkout` command does).

At any point, you can run `git status` to verify which branch you're on. It's incredibly important to always be aware of which git branch you're working on. Working on the wrong branch can lead to problems later on, so always try to do new work in a new branch based on the latest version of `master`.

## Step 2: Convert `Project1` to a private npm package (referred to here as "your project")

- [ ] Install Node 12 LTS in your ubuntu environment
- [ ] Install the Yarn package manager for Node
- [ ] Create a `package.json` file in your `Project1` folder using [`yarn init`](https://classic.yarnpkg.com/en/docs/creating-a-project/)
  - set `"private": true`
  - set `"license": "UNLICENSED"`

## Step 3: Add `Prettier` code formatter to `Project1`

[Prettier](https://prettier.io/) is an amazing tool, it will auto-format your code on every save (so long as you've installed the Prettier extension for your code editor) based on a customizable set of rules.

- [ ] Install Prettier as a _development_ dependency to your project, using `yarn` (_not_ `npm`)
- [ ] Install the Prettier extension for your code editor

**Comment by christiannaths** [comment]
@thjones88 Just a note, for any questions you have it's probably best to either ask here in the issue comments or in Slack

**Comment by thjones88** [comment]
Noted! Thanks.

**Comment by thjones88** [comment]
Do you generally push the new branch when you first start it so that it shows in the repo, or wait until make some type of change first?

**Comment by christiannaths** [comment]
Yeah, no sense pushing a branch with no changes.

I usually push a couple of times a day, after I've made a handful of commits. I kinda treat it like backup, like "what happens if my computer lights on fire right now" I don't want to lose any work, so pushing to GH helps mitigate that.

**Comment by thjones88** [comment]
Getting the correct version of node.js with npm is fine, but I want to install prettier via yarn?

**Comment by christiannaths** [comment]
Yep. There are two main package managers for Node: npm and yarn. Yarn is ahead of npm in almost every way, so install that and use `yarn` commands to install packages (like prettier)

It gets confusing because everybody refers to node packages as "npm packages", even though you can install them with any node package manager
